diff --git a/includes/ds.field_ui.inc b/includes/ds.field_ui.inc
index e485e4c..eba4f9d 100644
--- a/includes/ds.field_ui.inc
+++ b/includes/ds.field_ui.inc
@@ -2100,7 +2100,10 @@ function _ds_field_ui_core_fields($entity_type, $bundle, $view_mode, &$form, &$f
         if (!empty($summary)) {
           $summary .= '<br />';
         }
-        $summary .= 'Field template: ' . check_plain($functions[$field_function]) . '<br />';
+
+        if (isset($functions[$field_function])) {
+          $summary .= 'Field template: ' . check_plain($functions[$field_function]) . '<br />';
+        }
       }
 
       if (!empty($summary)) {
